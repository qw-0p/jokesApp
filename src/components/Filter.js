import React from 'react';
import { DropdownButton, Dropdown, Col } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';

import { filterList } from '../redux/ducks/jokes';

export const Filter = () => {
    const dispatch = useDispatch();
    const state = useSelector((state) => state.jokes.categories);
    const loading = useSelector((state) => state.jokes.loading);
    const submitHandler = (category) => {
        dispatch(filterList(category));
    };

    return (
        <Col>
            <DropdownButton id='dropdown-item-button' title='Filter' disabled={loading}>
                <Dropdown.Item as='button' onClick={() => submitHandler(null)}>
                    all
                </Dropdown.Item>
                {state.map(
                    (item) =>
                        item && (
                            <Dropdown.Item as='button' key={item} onClick={() => submitHandler(item)}>
                                {item}
                            </Dropdown.Item>
                        )
                )}
            </DropdownButton>
        </Col>
    );
};
