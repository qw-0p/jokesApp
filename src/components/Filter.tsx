import React from 'react';
import { DropdownButton, Dropdown, Col } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import { filterList } from '../redux/ducks/jokes';
import { RootState } from '../redux/store';


export const Filter: React.FC = () => {
    const dispatch = useDispatch();
    const state = useSelector((state: RootState): string[] => state.jokes.categories);
    const loading = useSelector((state: RootState): boolean => state.app.loading);
    const submitHandler = (event: any): void => {
        dispatch(filterList(event));
    };
    
    return (
        <Col>
            <DropdownButton id='dropdown-item-button' title='Filter' disabled={loading}>
                <Dropdown.Item as='button' onClick={(): void => submitHandler(null)}>
                    all
                </Dropdown.Item>
                {state.map(
                    (item: any): JSX.Element =>
                        item && (
                            <Dropdown.Item as='button' key={item} onClick={(): void => submitHandler(item)}>
                                {item}
                            </Dropdown.Item>
                        )
                )}
            </DropdownButton>
        </Col>
    );
};
