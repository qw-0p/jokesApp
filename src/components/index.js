export { Error } from './Error';
export { Filter } from './Filter';
export { Home } from './Home';
export { Item } from './Item';
export { JokesForm } from './JokesForm';
export { List } from './List';
export { Loader } from './Loader';
