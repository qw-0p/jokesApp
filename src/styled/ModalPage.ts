import styled from 'styled-components';

export const JokeText = styled.p`
    text-align: center;
    font-size: 1.75rem;
    font-weight: bold;
    color: #404040;
`;
