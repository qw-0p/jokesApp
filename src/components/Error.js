import React from 'react';
import { Row } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { ErrorPage } from '../styled/Error';

export const Error = () => (
    <ErrorPage>
        <Row>Page not found</Row>
        <Row>
            <Link to='/'>Home</Link>
        </Row>
    </ErrorPage>
);
