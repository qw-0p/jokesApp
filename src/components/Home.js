import React from 'react';
import { Col, Row } from 'react-bootstrap';
import { JokesForm } from './JokesForm';
import { Filter } from './Filter';
import { List } from './List';
import { Loader } from './Loader';
import { Wrapper } from '../styled/Home';

export const Home = () => (
    <Wrapper fluid>
        <Row>
            <Col>
                <Filter />
            </Col>
        </Row>
        <Row>
            <Col>
                <List />
                <Loader />
            </Col>
        </Row>
        <Row className='justify-content-center'>
            <JokesForm />
        </Row>
    </Wrapper>
);
