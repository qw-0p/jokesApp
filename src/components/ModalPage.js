import React from 'react';
import { Modal, Button } from 'react-bootstrap';

import { JokeText } from '../styled/ModalPage';

export const ModalPage = ({ show, joke, onHide }) => {
    return (
        <Modal show={show} centered size='lg'>
            <Modal.Body>
                <JokeText>{joke}</JokeText>
            </Modal.Body>
            <Modal.Footer>
                <Button onClick={() => onHide()}>Close</Button>
            </Modal.Footer>
        </Modal>
    );
};
