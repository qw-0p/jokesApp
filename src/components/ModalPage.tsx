import React from 'react';
import { Modal, Button } from 'react-bootstrap';

import { JokeText } from '../styled/ModalPage';

type ModalProps = {show: boolean, joke: string, onHide: () => void}

export const ModalPage: React.FC<ModalProps> = ({ show = false, joke = '', onHide }) => (
    <Modal show={show} centered size='lg'>
        <Modal.Body>
            <JokeText dangerouslySetInnerHTML={{ __html: joke }} />
        </Modal.Body>
        <Modal.Footer>
            <Button onClick={() => onHide()}>Close</Button>
        </Modal.Footer>
    </Modal>
);
