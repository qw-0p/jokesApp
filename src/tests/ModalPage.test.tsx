import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ModalPage, ModalProps } from '../components/ModalPage';

const props: ModalProps = {
    show: false,
    joke: '',
    onHide: () => {}
}

describe('ModalPage component', () => {
    it('snapshot', () => {
        const component = shallow( <ModalPage {...props} /> );
        expect(shallowToJson(component)).toMatchSnapshot();
    });
});