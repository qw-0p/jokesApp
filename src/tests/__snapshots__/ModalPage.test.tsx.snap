// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ModalPage component snapshot 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  centered={true}
  dialogAs={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "render": [Function],
    }
  }
  enforceFocus={true}
  keyboard={true}
  restoreFocus={true}
  show={false}
  size="lg"
>
  <ModalBody>
    <styled.p
      dangerouslySetInnerHTML={
        Object {
          "__html": "",
        }
      }
    />
  </ModalBody>
  <ModalFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      variant="primary"
    >
      Close
    </Button>
  </ModalFooter>
</Modal>
`;
